search-provider/eks-article-metadata-query-provider-dbus.xml

```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE node PUBLIC
+"-//freedesktop//DTD D-BUS Object Introspection 1.0//EN"
+"http://www.freedesktop.org/standards/dbus/1.0/introspect.dtd">
+
+<!--
+ This library is free software; you can redistribute it and/or
+ modify it under the terms of the GNU Lesser General Public
+ License as published by the Free Software Foundation; either
+ version 2 of the License, or (at your option) any later version.
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General
+ Public License along with this library; if not, see <http://www.gnu.org/licenses/>.
+-->
+<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
+  <interface name="com.endlessm.ArticleMetadata">
+    <method name="Query">
+      <arg type="as" name="Shards" direction="out" />
+      <arg type="aa{sv}" name="Results" direction="out" />
+      <arg type="a{sv}" name="Query" direction="in" />
```

Comment:
Maybe document the acceptable keys for this variant here

Reply:
Yep, something like this: https://gitlab.gnome.org/GNOME/gnome-shell/blob/master/data/org.gnome.ShellSearchProvider2.xml#L46

Reply:
Ah, we discussed on the phone explicitly specifying the parameters but I think now that might not actually work, considering that some of the query parameters are optional. I think we'll need to just document it for now.